engine/src/em-liburl.js

```diff
@@ -222,23 +222,17 @@ mergeInto(LibraryManager.library, {
                 type = 'error';
             }
 
-            // we want to return the raw data back to libURL
-            // however do as javascript assumes a DOMString and performs the
-            // appropriate char set conversions when returning the result
-            // thus we base64 encode on this side (and decode in libURL)
+			// To return the raw data back to libURL we use a FileReader
+			// to read the data to an ArrayBuffer then construct a
+			// Uint8Array (which will be converted to MCDataRef) from that.
             var xhr = this;
             if (this.responseType == 'blob') {
-                // the FileReader API allows us to convert blobs to base64
-                // the API prefixes base64 data with "data:text/plain;base64,"
-                // make sure we chop this off
                 var reader = new FileReader();
```

Comment:
I think it's possible to set the `responseType` of the request to `arraybuffer`, from which we can create a `Uint8Array` directly, removing the need for the `FileReader`. It shouldn't make any functional difference but is probably the slightly neater way to do things.